docker/multiarch.Dockerfile

```diff
@@ -0,0 +1,38 @@
+ARG SERVE_U=amd64
+ARG SERVE_R=alpine
+
+### Build ###
+FROM golang:1.11-alpine AS builder
+
+ARG BUILD_GOOS=linux
+ARG BUILD_GOARCH=amd64
+ENV CGO_ENABLED=0
+ENV BUILD_HOST=syncthing.net
+ENV BUILD_USER=docker
+
+WORKDIR /go/src/github.com/syncthing/syncthing
+COPY . .
+RUN go run build.go -no-upgrade -goos=$BUILD_GOOS -goarch=$BUILD_GOARCH build
+
+### Serve ###
+FROM $SERVE_U/$SERVE_R
+
+ARG QEMUARCH=amd64
+ENV PUID=1000 PGID=1000
+EXPOSE 8384 22000 21027/udp
+VOLUME ["/var/syncthing"]
+
+__MULTIARCH_COPY qemu-${QEMUARCH}-static /usr/bin/
+RUN apk add --update --no-cache ca-certificates su-exec
```

Comment:
I read some articles and now understand (I think) that at this point we will have an `apk` binary for arm (or whatever) and use the qemu here via a binfmt_misc handler that was  registered earlier to run it. Also, to run the subsequent `rm`... This could really use a commit of some kind here, because it's totally non-obvious unless you're familiar with the background on multiarch docker images.

Reply:
With "commit" do you mean that we should add a comment explaining it there?